File: java/src/com/mogreet/mercury/system/Ping.java
/**
* Copyright 2011 Nicolas Payot
*
* This file is part of Mercury.
*
* Mercury is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* Mercury is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with Mercury.  If not, see <http://www.gnu.org/licenses/>.
*/

package com.mogreet.mercury.system;

import com.mogreet.mercury.*;
import org.w3c.dom.*;

/**
* The Ping object contains the response from a {@link Mercury#ping} request to the Moms API.
*/
public class Ping extends Response {
	
	/**
	* Creates a new Ping response object.
	* @param xmlDoc XML DOM document that is returned by all the requests.
	*/
	public Ping(Document xmlDoc) {
		super(xmlDoc);
		if (!super.responseIsValid())
			throw new RuntimeException(super.getMessage());
	}
}

File: dotNet/MogreetSDK/MogreetSDK/system/Ping.cs
using System;
using System.Xml;
using System.Security;

namespace MogreetSDK.system
{
    /// <summary>
    /// The Ping object contains the response from a <see cref="MogreetSDK.Mercury.ping"/> request to the Moms API.
    /// </summary>
    public class Ping : Response
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SDK.Ping"/> class.
        /// </summary>
        /// <param name='xmlDoc'>
        /// Xml document that is returned by all the requests.
        /// </param>
        /// <exception cref='Exception'>
        /// Represents errors that occur during application execution.
        /// </exception>
        public Ping(XmlDocument xmlDoc)
            : base(xmlDoc)
        {
            if (!(base.responseIsValid()))
                throw new Exception(base.getMessage());
        }
    }
}
